eng/Versions.props

```diff
@@ -43,8 +43,8 @@
   </ItemGroup>
   <PropertyGroup>
     <!-- For source generator support we need to target a pinned version in order to be able to run on older versions of Roslyn -->
-    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>3.8.0</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
-    <MicrosoftCodeAnalysisVersion>3.8.0</MicrosoftCodeAnalysisVersion>
+    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>3.9.0</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
```

Comment:
cc @ericstj, upgraded the compiler version here so I could use [`ITypeSymbol.IsRecord`](https://docs.microsoft.com/dotnet/api/microsoft.codeanalysis.itypesymbol.isrecord?view=roslyn-dotnet-3.10.0#Microsoft_CodeAnalysis_ITypeSymbol_IsRecord) in the generator implementation.